### PR TITLE
TSERV-194: update testserver client library

### PR DESCRIPTION
**Pull request by veinSmart**
update testserver client library with callbackurl

### CODE REVIEW

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 1]
```diff
@@ -301,14 +301,15 @@ public void setBasePath(String basePath) {
...
-                                           String endpoint, boolean async, HttpBasicAuth auth) throws ApiException {
+                                           String endpoint, String callBackUrl, boolean async, HttpBasicAuth auth) throws ApiException {
         if (!swaggerFile.exists()) {
             throw new ApiException(404, "File [" + swaggerFile.toString() + "] not found");
         }
         setAuthentication(auth);
         List<Pair> queryParams = new ArrayList<>();
         queryParams.add(new Pair("async", String.valueOf(false)));
         queryParams.add(new Pair("endpoint", endpoint));
+        queryParams.add(new Pair("callBack", callBackUrl));
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: callBack -> callback

Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: Also, you can add it only if it's not null/empty.

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 2]
```diff
@@ -328,6 +331,9 @@ public ProjectResultReport postSwagger(URL swaggerApiURL, String endpoint, boole
         List<Pair> queryParams = new ArrayList<>();
         queryParams.add(new Pair("async", String.valueOf(async)));
         queryParams.add(new Pair("endpoint", endpoint));
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: endpoint can be null also - should we have a similar check?

Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: True, @veinSmart can you please add the null check for endpoint also while you are at it ?

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 3]
```diff
@@ -301,14 +301,19 @@ public void setBasePath(String basePath) {
...
-                                           String endpoint, boolean async, HttpBasicAuth auth) throws ApiException {
+                                           String endpoint, String callBackUrl, boolean async, HttpBasicAuth auth) throws ApiException {
         if (!swaggerFile.exists()) {
             throw new ApiException(404, "File [" + swaggerFile.toString() + "] not found");
         }
         setAuthentication(auth);
         List<Pair> queryParams = new ArrayList<>();
         queryParams.add(new Pair("async", String.valueOf(false)));
-        queryParams.add(new Pair("endpoint", endpoint));
+        if(StringUtils.isNotEmpty(callBackUrl)) {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: should be endpoint :-)

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 4]
```diff
@@ -319,15 +324,20 @@ public ProjectResultReport postSwagger(File swaggerFile, SwaggerApiValidator.Swa
...
+    public ProjectResultReport postSwagger(URL swaggerApiURL, String endpoint, String callBackUrl, boolean async, HttpBasicAuth auth)
             throws ApiException {
         if (swaggerApiURL == null) {
             throw new ApiException(404, "Swagger API URL is null.");
         }
         setAuthentication(auth);
         List<Pair> queryParams = new ArrayList<>();
         queryParams.add(new Pair("async", String.valueOf(async)));
-        queryParams.add(new Pair("endpoint", endpoint));
+        if(StringUtils.isNotEmpty(callBackUrl)) {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: same here :-)

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java (hunk 3 above)
Comment (veinSmart) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: LOL....sorry for that one...

**Comment by olensmar** [comment]
looks good now - thanks!

**Pull request merged**